lib/spack/spack/database.py

```diff
@@ -203,19 +203,30 @@ def __init__(self, root, db_dir=None):
             mkdirp(self._db_dir)
 
         # initialize rest of state.
-        self.lock = Lock(self._lock_path)
+        self.db_lock_timeout = (
+            spack.config.get('config:db_lock_timeout') or _db_lock_timeout)
+        self.package_lock_timeout = (
+            spack.config.get('config:package_lock_timeout') or None)
+        tty.debug('DATABASE LOCK TIMEOUT: {0}s'.format(
+                  str(self.db_lock_timeout)))
+        timeout_format_str = ('{0}s'.format(str(self.package_lock_timeout))
+                              if self.package_lock_timeout else 'No timeout')
+        tty.debug('PACKAGE LOCK TIMEOUT: {0}'.format(
```

Comment:
In addition to these debug statements, if it's easy, could you add some debug info to the "acquired" `tty.debug` call?  It would be good for debugging if it said "acquired after XX.XXs and XX attempts" instead of just "acquired"